packages/services/api/src/modules/rate-limit/providers/rate-limit.provider.ts

```diff
@@ -1,20 +1,21 @@
+import { createClient } from 'fets';
 import { Inject, Injectable, Scope } from 'graphql-modules';
 import type { RateLimitApi, RateLimitApiInput } from '@hive/rate-limit';
-import { createTRPCProxyClient, httpLink } from '@trpc/client';
-import { fetch } from '@whatwg-node/fetch';
 import { HiveError } from '../../../shared/errors';
 import { sentry } from '../../../shared/sentry';
 import { Logger } from '../../shared/providers/logger';
 import type { RateLimitServiceConfig } from './tokens';
 import { RATE_LIMIT_SERVICE_CONFIG } from './tokens';
 
+type CheckRateLimitInput = RateLimitApiInput['/check-rate-limit']['post']['json'];
+
 @Injectable({
   global: true,
   scope: Scope.Singleton,
 })
 export class RateLimitProvider {
   private logger: Logger;
-  private rateLimit;
+  private rateLimit: RateLimitApi['__client'] | null;
```

Comment:
Can we consider improving the API of this? the `[__client]` feels like a hack?

Reply:
I forgot to change it, we have `RouterClient` type.

Reply:
Done! 👍